app/models/reply.rb

```diff
@@ -12,10 +12,11 @@ class Reply < ApplicationRecord
   validate :author_can_write_in_post, on: :create
   audited associated_with: :post, except: :reply_order, update_with_comment_only: false
 
-  after_create :notify_other_authors, :destroy_draft, :update_active_char, :set_last_reply, :update_post, :update_post_authors
-  after_save :update_flat_post
+  after_create :notify_other_authors, :destroy_draft, :update_active_char, :set_last_reply, :update_post
+  after_create_commit :update_post_authors # this can generate notification jobs
```

Comment:
Is there a way to get just the notification job into an after_commit instead?

Reply:
...probably, if it you split it out into it's own method? That seems like overkill to me but you _could_

Reply:
Exceptions raised in `commit` blocks are basically ignored – since they're after a commit, they can't influence whether the base model gets saved or not, meaning we risk an error when creating a post author here and would have an inconsistency in our PostAuthor models.

I think it'd be better to leave this in `after_create` (potentially notifying someone erroneously) rather than putting it into `after_create_commit` (potentially having PostAuthor inconsistent with the post), unless we can split out just the notification logic to the `after_commit`.